# TODO: Make the global params actually "global" so they don't need to be defined in each node

# Feature Extraction Params
feature_extraction_node:
    ros__parameters:
        use_sim_time: false
        PROJECT_NAME: ""

        imu_topic: "/imu/data"
        laser_topic: "/lidar/scan"  #velodyne_points  #/livox/lidar /lidar/scan
        odom_topic: "/dummy"
        depthup_topic: "/rs_up/depth/cloud_filtered"
        depthdown_topic: "/rs_down/depth/cloud_filtered"
        world_frame: "map"
        world_frame_rot: "map_rot"
        sensor_frame: "sensor"
        sensor_frame_rot: "sensor_rot"

        scan_line: 4 #16 64
        sensor: "livox"
        mapping_skip_frame: 1
        blindFront: 0.1
        blindBack: -1.0
        blindLeft: 0.1
        blindRight: -0.1
        use_imu_roll_pitch: false
        use_dynamic_mask: false
        use_up_realsense_points: false
        use_down_realsense_points: false
        min_range: 0.2    #<!-- remove too closed points -->
        max_range: 130.
        skip_realsense_points: 3
        provide_imu_laser_extrinsic: true
        provide_point_time: 1
        debug_view: false
        
        imu_acc_x_limit: 1.0
        imu_acc_y_limit: 1.0
        imu_acc_z_limit: 1.0

# Laser Mapping Params
laser_mapping_node:
    ros__parameters:
        use_sim_time: false
        PROJECT_NAME: ""

        imu_topic: "imu/data"
        laser_topic: "velodyne_points"
        odom_topic: "integrated_to_init5"
        depthup_topic: "/rs_up/depth/cloud_filtered"
        depthdown_topic: "/rs_down/depth/cloud_filtered"
        world_frame: "map"
        world_frame_rot: "map_rot"
        sensor_frame: "laser"
        sensor_frame_rot: "sensor_rot"

        mapping_line_resolution: 0.1
        mapping_plane_resolution: 0.2
        max_iterations: 5
        debug_view: false
        enable_ouster_data: false
        publish_only_feature_points: false
        use_imu_roll_pitch: false
        start_from_previous_map: false
        max_surface_features: 2000
        velocity_failure_threshold: 30.0
        provide_imu_laser_extrinsic: true
        auto_voxel_size: true
        forget_far_chunks: false
        visual_confidence_factor: 1.0
        pos_degeneracy_threshold: 1.0
        ori_degeneracy_threshold: 1.0

        local_mode: false
        read_pose_file: false
        init_x: 0.0
        init_y: 0.0
        init_z: 0.0
        init_roll: 0.0
        init_pitch: 0.0
        init_yaw: 0.0

# IMU Preintegration Params
imu_preintegration_node:
    ros__parameters:
        use_sim_time: false
        PROJECT_NAME: ""
        sensor: "livox"
        imu_topic: "/imu/data"
        laser_topic: "/lidar/scan"
        odom_topic:  "integrated_to_init"
        depthup_topic: "/rs_up/depth/cloud_filtered"
        depthdown_topic: "/rs_down/depth/cloud_filtered"
        world_frame: "map"
        world_frame_rot: "map_rot"
        sensor_frame: "sensor"
        sensor_frame_rot: "sensor_rot"

        lidar_correction_noise: 0.01
        smooth_factor: 0.9
        provide_imu_laser_extrinsic: true 
        acc_n: 3.9939570888238808e-03          # accelerometer measurement noise standard deviation. #0.2
        gyr_n: 1.5636343949698187e-03         # gyroscope measurement noise standard deviation.     #0.05
        acc_w: 6.4356659353532566e-05          # accelerometer bias random work noise standard deviation.  #0.02
        gyr_w: 3.5640318696367613e-05       # gyroscope bias random work noise standard deviation.     #4.0e-5
        g_norm: 9.80511    # 0 means you need to provide both imu-camera and laser-camera extrinsic
        use_imu_roll_pitch: false
        imu_acc_x_limit: 0.5
        imu_acc_y_limit: 0.2
        imu_acc_z_limit: 0.4
